### ISSUE TITLE
native uuid support not working correctly

### ISSUE THREAD

**Issue by hapm** [m32/sqlalchemy-tds#11]
I tried to use a `uuid.UUID` in one of my orm classes but ran into the following error:

```
  File "[...]\Lib\site-packages\sqlalchemy\sql\sqltypes.py", line 3638, in process
    value = _python_UUID(value)
            ^^^^^^^^^^^^^^^^^^^
  File "[...]\Lib\uuid.py", line 175, in __init__
    hex = hex.replace('urn:', '').replace('uuid:', '')
          ^^^^^^^^^^^
AttributeError: 'UUID' object has no attribute 'replace'
```

After investigating I found that the dialect seems to have `supports_native_uuid = False`.

Adding this to the [`def __init__`of `MSDialect_pytds`](https://github.com/m32/sqlalchemy-tds/blob/75cfae04ba19c340f702c2d62a20b94609239b38/sqlalchemy_pytds/dialect.py#L136)  solved it for me:
```
self.supports_native_uuid = True

**Comment by m32** [comment]
I added missing settings for decimal and uuid.
Check how your program works without additional modifications with the current version of the library downloaded from github and let me know.
Before I publish another version in pypi.

**Comment by hapm** [comment]
Thanks for the fast reply, will take me a while to test, because I'm not at work for the next 4 weeks. Btw: I didn't see any additional commits, are the changes pushed already?

**Comment by m32** [comment]
I forgot about that one - now pushed

**Comment by gordthompson** [comment]
I have verified the fixes in 283b66f using the UuidTest test in the test suite.

Before 283b66f:

```
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_literal_nonnative_text PASSED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_literal_nonnative_uuid PASSED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_literal_text FAILED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_literal_uuid FAILED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_uuid_returning FAILED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_uuid_round_trip FAILED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_uuid_text_round_trip FAILED
```

After 283b66f:

```
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_literal_nonnative_text PASSED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_literal_nonnative_uuid PASSED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_literal_text PASSED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_literal_uuid PASSED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_uuid_returning PASSED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_uuid_round_trip PASSED
test/test_suite.py::UuidTest_mssql+pytds_15_0_15_183::test_uuid_text_round_trip PASSED
```

**Comment by m32** [comment]
closing as resolved
